marconi-sidechain/src/Marconi/Sidechain/Api/Query/Indexers/MintBurn.hs

```diff
@@ -79,7 +88,21 @@ queryByPolicyAndAssetId env policyId assetId slotNo txId = do
       let q = MintBurn.QueryBurnByAssetId policyId assetId slotNo txId
       res <- runExceptT $ Storable.query indexer q
       case res of
-        Right (MintBurnResult mintBurnRows) -> pure $ Right $ toAssetIdTxResult <$> mintBurnRows
+        Right (MintBurnResult mintBurnRows) -> do
+          case slotNo of
+            Nothing -> do
+              -- The user didn't provide an upper bound slot number. Therefore, we query the latest
+              -- block number of the local node.
+              blockNoE <- queryCurrentNodeBlockNo env
+              pure $ do
+                blockNo <- blockNoE
+                pure $ toAssetIdTxResult blockNo <$> mintBurnRows
+            Just s -> do
+              -- The user provided an upper bound slot number. Therefore, we convert the provided
+              -- slot number to a block number by using the indexer which indexes that information.
+              let addressUtxoIndexerEnv = env ^. sidechainEnvIndexers . sidechainAddressUtxoIndexer
+              blockNo <- Utxo.queryBlockNoAtSlotNo addressUtxoIndexerEnv s
```

Comment:
I don't get why we don't always consider the latest blockNo to assess stability: if I ask for something with an upperbound at slot 1000, it we're at slot 1000000, it will be stable. So I would think that only the current block number is relevant.

Reply:
That was a direct request from Sidechain team. When querying for burning events, they want to know what was stable by using as reference the input slotNo.